lib/reply.js

```diff
@@ -216,35 +211,32 @@ Reply.prototype.removeHeader = function (key) {
   return this
 }
 
-Reply.prototype.header = function (key, value) {
-  const _key = key.toLowerCase()
-
-  // default the value to ''
-  value = value === undefined ? '' : value
+Reply.prototype.header = function (key, value = '') {
+  key = key.toLowerCase()
 
-  if (this[kReplyHeaders][_key] && _key === 'set-cookie') {
+  if (this[kReplyHeaders][key] && key === 'set-cookie') {
     // https://tools.ietf.org/html/rfc7230#section-3.2.2
-    if (typeof this[kReplyHeaders][_key] === 'string') {
-      this[kReplyHeaders][_key] = [this[kReplyHeaders][_key]]
+    if (typeof this[kReplyHeaders][key] === 'string') {
+      this[kReplyHeaders][key] = [this[kReplyHeaders][key]]
     }
+
     if (Array.isArray(value)) {
-      Array.prototype.push.apply(this[kReplyHeaders][_key], value)
+      this[kReplyHeaders][key].push(...value)
     } else {
-      this[kReplyHeaders][_key].push(value)
+      this[kReplyHeaders][key].push(value)
     }
   } else {
-    this[kReplyHeaders][_key] = value
+    this[kReplyHeaders][key] = value
   }
+
   return this
 }
 
 Reply.prototype.headers = function (headers) {
-  const keys = Object.keys(headers)
-  /* eslint-disable no-var */
-  for (var i = 0; i !== keys.length; ++i) {
-    const key = keys[i]
-    this.header(key, headers[key])
+  for (const headerName in headers) {
```

Comment:
There is a _significant_ difference between `Object.keys` and `for...in`:

`for...in` https://262.ecma-international.org/13.0/#sec-createforiniterator:
> It is used to create a For-In Iterator object which iterates over the ***own and inherited*** enumerable string properties of object in a specific order.

`Object.keys` https://262.ecma-international.org/13.0/#sec-object.keys:
> 2. 2. Let nameList be ? ***[EnumerableOwnPropertyNames](https://262.ecma-international.org/13.0/#sec-enumerableownpropertynames)(obj, key)***.

In short, the `Object.keys` implementation _does not_ enumerate properties from higher up in the prototype chain. It only iterates the actual headers object properties.